### ISSUE TITLE
Contributing to the Upstream Kubernetes Code - Missleading/Wrong Docs/Information

### ISSUE THREAD

**Issue by iamNoah1** [kubernetes/website#32718]
Following the Contributing to the Upstream Kubernetes Code, I have pretty much the same use case as mentioned. Updating the Comment section for IngressClassName to improve the API docs. Thus I made changes on `staging/src/k8s.io/api/networking/v1/types.go` and ran the scripts: `hack/update-generated-swagger-docs.sh`, `hack/update-openapi-spec.sh` and `hack/update-generated-protobuf.sh`. 

My output looks slighty different than the one showed on the Docs Page: 

	modified:   pkg/generated/openapi/zz_generated.openapi.go
	modified:   staging/src/k8s.io/api/networking/v1/generated.proto
	modified:   staging/src/k8s.io/api/networking/v1/types_swagger_doc_generated.go

Specifically I am missing changes on `api/openapi-spec/swagger.json` here and wondering about the `pkg/generated/openapi/zz_generated.openapi.go`. Is it maybe just an outdated part of the doc? Or am I doing something wrong?

**Comment by tengqm** [comment]
Maybe `hack/update-generated-swagger-docs.sh` is another script to be triggered.

**Comment by sftim** [comment]
https://kubernetes.io/docs/contribute/generate-ref-docs/contribute-upstream/ predates https://k8s.dev/docs/

If we can, it'd be good to move this whole process into https://k8s.dev/docs/ and then add redirects (or change hyperlinks).

**Comment by iamNoah1** [comment]
> Maybe `hack/update-generated-swagger-docs.sh` is another script to be triggered.

I ran that script as well.

**Comment by iamNoah1** [comment]
> https://kubernetes.io/docs/contribute/generate-ref-docs/contribute-upstream/ predates https://k8s.dev/docs/
> 
> If we can, it'd be good to move this whole process into https://k8s.dev/docs/ and then add redirects (or change hyperlinks).
I see, so can I assume that https://k8s.dev/docs/ will be the new source of truth for contribution docs? 

However, my question is still if I am doing something wrong or if the docs are out of date? If they are out of date I would love to update them accordingly.

**Comment by sftim** [comment]
If you have contributed to upstream Kubernetes and want to ensure you've done all the docs generation steps, I think `make update` is canonical. It covers everything (and uses a lot of compute).

**Comment by sftim** [comment]
> can I assume that https://k8s.dev/docs/ will be the new source of truth for contribution docs

Bear in mind that this documentation is maintained by Kubernetes contributors and that means, mostly by volunteers. The intent is certainly for https://k8s.dev/docs/ to be canonical.

**Comment by iamNoah1** [comment]
> > can I assume that https://k8s.dev/docs/ will be the new source of truth for contribution docs
> 
> Bear in mind that this documentation is maintained by Kubernetes contributors and that means, mostly by volunteers. The intent is certainly for https://k8s.dev/docs/ to be canonical.

Don't know what exactly you want to tell me with that piece information 😅. However I found the root cause why the output differed for me and swagger.json was not generated. Running `./hack/update-openapi-spec.sh` was not successful and I did not pay enough attention. The script is looking for `.../kubernetes/_output/local/bin/darwin/arm64/kube-apiserver` which does not exist when running on a Apple Silicon machine. 

I will create an issue for that. After running everything on a ubuntu amd vm, at least the swagger.json was generated but the output still looks slightly different, so I will create a PR to update the docs :)
